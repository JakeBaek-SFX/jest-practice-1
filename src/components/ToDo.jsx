import React from 'react';

function ToDo({toDoListItem}) {
  const { id, title, isCompleted } = toDoListItem;

  return (
    <div data-testid={`to-do-${id}`}>
      <h1>{title}</h1>
      {isCompleted && <strike>completed!</strike>}
      <div>hi</div>
    </div>
  );
}

export default ToDo;