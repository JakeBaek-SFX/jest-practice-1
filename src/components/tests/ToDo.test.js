import { render, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import ToDo from '../ToDo';

afterEach(() => {
  cleanup();
})

test('component test for ToDo', () => {
  const tempList = {id: 1, title: 'make money', isCompleted: true};
  render(<ToDo toDoListItem={tempList}/>);
  const toDoElement = screen.getByTestId(`to-do-${tempList.id}`);
  expect(toDoElement).toBeInTheDocument();
  expect(toDoElement).toHaveTextContent(tempList.title);
});

test('matches snapshot', () => {
  const tempList = {id: 1, title: 'make money', isCompleted: true};
  const tree = renderer.create(<ToDo toDoListItem={tempList}/>).toJSON();
  expect(tree).toMatchSnapshot();
})