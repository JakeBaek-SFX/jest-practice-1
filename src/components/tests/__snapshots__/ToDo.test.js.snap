// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  data-testid="to-do-1"
>
  <h1>
    make money
  </h1>
  <strike>
    completed!
  </strike>
  <div>
    hi
  </div>
</div>
`;
